typings/objection/index.d.ts

```diff
@@ -336,24 +336,15 @@ declare module "objection" {
     insertWithRelated: Insert<T>;
     insertWithRelatedAndFetch: InsertGraphAndFetch<T>
 
-    /**
-     * @return a Promise of the number of updated rows
-     */
-    update(modelOrObject: Partial<T>): QueryBuilderSingle<number>;
+    update(modelOrObject: Partial<T>): QueryBuilder<T>;
     updateAndFetch(modelOrObject: Partial<T>): QueryBuilderSingle<T>;
     updateAndFetchById(id: Id, modelOrObject: Partial<T>): QueryBuilderSingle<T>;
 
-    /**
-     * @return a Promise of the number of patched rows
-     */
-    patch(modelOrObject: Partial<T>): QueryBuilderSingle<number>;
+    patch(modelOrObject: Partial<T>): QueryBuilder<T>;
```

Comment:
Thanks for the catch, I agree it shouldn't be `QueryBuilderSingle<number>`. It seems like this should be `QueryBuilderSingle<T>`. If I'm remembering correctly, I made it a number because the docs say

> NOTE: The return value of the query will be the number of affected rows. If you want to patch a single row and retrieve the patched row as a result, 

and I didn't think about subsequent useful calls to the queryBuilder like `.where`.